app/script/media/MediaStreamHandler.js

```diff
@@ -209,11 +227,9 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
       : Promise.resolve(mediaStreamInfo);
 
     return replacePromise.then(updateMediaStreamInfo => {
-      const mediaType = !updateMediaStreamInfo.replaced ? type : undefined;
-
-      this._setStreamState(updateMediaStreamInfo);
-      this._releaseMediaStream(this.localMediaStream(), mediaType);
-      this.localMediaStream(updateMediaStreamInfo.stream);
+      return updateMediaStreamInfo.replaced
+        ? replaceMediaStream(this.localMediaStream(), updateMediaStreamInfo)
+        : replaceMediaTracks(this.localMediaStream(), updateMediaStreamInfo.stream);
```

Comment:
We can also move this to the methods within the `updateJoinedCall`. In `_replaceMediaStream` and `_replaceMediaTrack` we already know how we have to update the local media stream. Doing it out here is legacy and we can improve this.

Reply:
:+1:

Reply:
Oh wait, we also need to do that in the case where the call is not yet joined, isn't it? So it seems legit that it is in the `replaceMediaStream` method, isn't it?